PR 62

ericsorenson opened the pull request:
### Stories & Bugs

- [7748 BUG: Cover sheet not being applied; page count is 0 (recurring?)](https://github.com/flexion/ef-cms/issues/7748) 
- [7717 BUG: IRS seeing wrong service indicator on primary petitioner before being associated with case](https://github.com/flexion/ef-cms/issues/7717) 
- [7700 BUG: Draft document title / description does not display in attach document dropdown on messages ](https://github.com/flexion/ef-cms/issues/7700) 
- [7574 Admissions Clerk: Ability to Update Practitioner Email Address](https://github.com/flexion/ef-cms/issues/7574) 
- [7501 BUG: Unexpected log out during order drafting](https://github.com/flexion/ef-cms/issues/7501) 
- [7406 Practitioner: Ability to Edit Email Address](https://github.com/flexion/ef-cms/issues/7406) 
- [6239 Court: Display Bench Opinions on Today's Opinions (Tom)](https://github.com/flexion/ef-cms/issues/6239) 
- [2223 Petitioner: Edit email address](https://github.com/flexion/ef-cms/issues/2223) 

### Refactoring
- https://trello.com/c/QU9S78OF
- https://trello.com/c/eWIAmFqM
- https://trello.com/c/ETTjuTfn
- https://trello.com/c/jnopE9E6
- https://trello.com/c/M50fKtPT
- https://trello.com/c/3RvvTYJ9
- https://trello.com/c/zTOTgPHq
- https://trello.com/c/5VzV3XPi
- https://trello.com/c/t4JEf9v8
- https://trello.com/c/AW6CG955
- https://trello.com/c/BcRg2BzP
- https://trello.com/c/rWaUmTEs
- https://trello.com/c/S7Ddbwom
- https://trello.com/c/ys324LNe
- https://trello.com/c/Ps2e0cj1
- https://trello.com/c/Paoqf1Tp
- https://trello.com/c/fvEIAiqa

### Migrations
- 17
- 18
- 19
- 20

mrinsin committed:
Merge pull request #7718 from flexion/204-action-tests

204 action test coverage

rachaelparris committed:
6239: refactoring; move bench opinion event code into opinion event codes

rachaelparris committed:
6239: DOD integration tests [skip ci]

codyseibert committed:
7501 - adding unit tests for various actions and sequences

rachaelparris committed:
6239 test fix

rachaelparris committed:
Merge pull request #7723 from flexion/6239-ost-todays-orders

6239: bench opinions

rachaelparris committed:
Remove honeybadger references

ericsorenson committed:
Merge pull request #7707 from flexion/disable-search-engine-indexing

Disable search engine indexing

kkoskelin committed:
7501: pr comments

restoring accidental deletion; addressing jsdocs per PR comments

rachaelparris committed:
Merge pull request #7724 from flexion/delete-honeybadger-references

Remove honeybadger references

matthopson committed:
Merge remote-tracking branch 'origin/develop' into 7501-bug-tabbed-idle-logout

matthopson committed:
Merge branch '7501-bug-tabbed-idle-logout' of github.com:flexion/ef-cms into 7501-bug-tabbed-idle-logout

rachaelparris committed:
6239: DOD - finish up integration tests; npm update

matthopson committed:
Merge pull request #7722 from flexion/7501-bug-tabbed-idle-logout

7501 - Tabbed Idle Logout Bug

rachelschneiderman committed:
7406: Remove 'My Contact Information' from private practitioner dashboard

rachelschneiderman committed:
7406: Remove 'My Contact Information' from irs practitioner dashboard

rachaelparris committed:
Merge branch 'develop' into 6239-dod

mrinsin committed:
7698 ensuring that all practitioner association document type titles do not get overwritten

rachelschneiderman committed:
7700: WIP properly compute draft document title for message modal [skip ci]

rachaelparris committed:
attempting an integration test fix

rachaelparris committed:
6239: address pr feedback

rachaelparris committed:
Merge pull request #7727 from flexion/7698-ea-qc-bug

7698 BUG: Unable to QC Entry of Appearance, Substitution of Counsel, or Limited Entry of Appearance

rachaelparris committed:
7700: add tests for draft document title bug

rachelschneiderman committed:
Update bug issue template, remove references to flexion prod and replace with court migration env

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 6239-dod

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7700-message-draft-doc-title-bug

rachelschneiderman committed:
Merge branch '7700-message-draft-doc-title-bug' of https://github.com/flexion/ef-cms into 7700-message-draft-doc-title-bug

rachaelparris committed:
Merge pull request #7729 from flexion/7406-remove-contact-info-from-dashboard

7406: Remove contact info from private/irs practitioner dashboard

rachaelparris committed:
Merge branch 'develop' into 6239-dod

rachaelparris committed:
Merge branch '6239-dod' of https://github.com/flexion/ef-cms into 6239-dod

rachaelparris committed:
Merge branch 'develop' into 7700-message-draft-doc-title-bug

rachaelparris committed:
Merge branch 'develop' into update-bug-template

ericsorenson committed:
migration env doesn't need smoketests

ericsorenson committed:
run jest tests in the same order every time

matthopson committed:
7501 - WIP set env vars for session timeout

rachaelparris committed:
downgrade babel

rachaelparris committed:
revert aws-sdk upgrade

matthopson committed:
Beefing up integration tests around trial session creation

rachaelparris committed:
7574: remove alternate email

matthopson committed:
7501 - Update setup.md

rachelschneiderman committed:
6239: update package-lock

rachelschneiderman committed:
6239: Attempt to lock down babel versioning

mrinsin committed:
ci build fixes

matthopson committed:
Merge pull request #7734 from flexion/7501-env-session-timeout

7501 env session timeout

mrinsin committed:
Merge branch '6239-dod' of https://github.com/flexion/ef-cms into 6239-dod

# Conflicts:
#	package-lock.json
#	package.json

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into integration-test-updates

mrinsin committed:
Merge branch '6239-dod' of https://github.com/flexion/ef-cms into 7700-message-draft-doc-title-bug

kkoskelin committed:
integration-test update

memory bump; debug docket number in test

mrinsin committed:
updating packages

mrinsin committed:
Merge branch '6239-dod' of https://github.com/flexion/ef-cms into update-bug-template

mrinsin committed:
Merge pull request #7726 from flexion/6239-dod

6239: DOD - integration tests; npm update

mrinsin committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into update-bug-template

mrinsin committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7700-message-draft-doc-title-bug

mrinsin committed:
updating packages

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into integration-test-updates

kkoskelin committed:
recreated package-lock

mrinsin committed:
Merge pull request #7732 from flexion/update-bug-template

Update bug issue template: remove references to flexion prod and replace with court migration env

mrinsin committed:
Merge pull request #7731 from flexion/7700-message-draft-doc-title-bug

Bug 7700: Draft document title / description does not display in attach document dropdown on messages

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into integration-test-updates

# Conflicts:
#	package-lock.json

kkoskelin committed:
updated package-lock

ericsorenson committed:
Merge branch 'develop' into migration-env-doesnt-need-smoke-tests

kkoskelin committed:
revert memory change for integration tests

ericsorenson committed:
add jest sequencer

rachaelparris committed:
Merge branch 'develop' into 7574-alternate-email

ericsorenson committed:
disable verbose mode

matthopson committed:
Address issues in pa11y tests

rachaelparris committed:
7574: update package-lock

rachelschneiderman committed:
7574: Copy package-lock directly from develop

ericsorenson committed:
Merge pull request #7735 from flexion/migration-env-doesnt-need-smoke-tests

Migration env doesnt need smoke tests, add jest sequencer

rachelschneiderman committed:
Merge branch '7574-alternate-email' of https://github.com/flexion/ef-cms into 7574-alternate-email

# Conflicts:
#	package-lock.json

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-alternate-email

# Conflicts:
#	package-lock.json

rachelschneiderman committed:
7574: Try downgrading local npm

rachelschneiderman committed:
7574: Try downgrading local npm to a lower version

rachelschneiderman committed:
Merge pull request #7736 from flexion/7574-alternate-email

7574: remove alternate email from practitioner

rachelschneiderman committed:
7574: WIP Adding Login & Service Email form to edit practitioner page

rachelschneiderman committed:
7574: WIP Adding Login & Service Email form to edit practitioner page with hardcoded data

rachelschneiderman committed:
7574: Edit practitioner user, Add new section labelled “Login & Service email”

rachaelparris committed:
7574: remove a couple of unused imports

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into integration-test-updates

# Conflicts:
#	package-lock.json

kkoskelin committed:
Merge branch 'integration-test-updates' of github.com:flexion/ef-cms into integration-test-updates

kkoskelin committed:
merge develop and re-generate package-lock

rachaelparris committed:
7574: only display login and service email form when editing a practitioner

mrinsin committed:
WIP 7574 adding validation for new email [skip ci]

rachaelparris committed:
Merge pull request #7737 from flexion/7574-add-login-service-email-form

7574: Add new section labelled “Login & Service email”

ericsorenson committed:
test waiting after login on CircleCI

ericsorenson committed:
Merge branch 'develop' into migration-env-doesnt-need-smoke-tests

rachelschneiderman committed:
7574: Consolidate updating email validation between Practitioner and NewPractitioner entities

ericsorenson committed:
Merge pull request #7738 from flexion/migration-env-doesnt-need-smoke-tests

Experimental: Get Integration tests moving while we look for the missing awaits

rachaelparris committed:
7574: check email availability

rachaelparris committed:
Merge branch 'develop' into login-email-change-validation

matthopson committed:
Merge remote-tracking branch 'origin/develop' into integration-test-updates

rachaelparris committed:
increase integration test timeout

ericsorenson committed:
try using a large instance for integration tests

rachelschneiderman committed:
Merge pull request #7739 from flexion/login-email-change-validation

7574: Login email change validation

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-check-email-availability

ericsorenson committed:
Merge pull request #7741 from flexion/migration-env-doesnt-need-smoke-tests

Try using a large instance for integration tests

ericsorenson committed:
Merge branch 'develop' into integration-test-updates

rachelschneiderman committed:
Merge pull request #7740 from flexion/7574-check-email-availability

7574: check email availability when editing practitioner email address

rachelschneiderman committed:
7574: Verification modal

klohman committed:
Update release-notes.md

mrinsin committed:
7574 renaming action to be more accurate

matthopson committed:
Update timeout value in readme to real integer

mrinsin committed:
7574 WIP updating email in API [skip ci]

klohman committed:
Update release-notes.md

matthopson committed:
Merge remote-tracking branch 'origin/develop' into integration-test-updates

ericsorenson committed:
Merge remote-tracking branch 'upstream/staging' into from-usataxcourt

codyseibert committed:
7574 - working on adding more unit tests

ericsorenson committed:
upgrades

rachaelparris committed:
7574: edit practitioner detail UI - pending email and email not provided

rachaelparris committed:
remove duplicate export in entity constants

rachaelparris committed:
increase test timeouts set to 10000ms to 30000ms

rachelschneiderman committed:
7475: Update API to handle pending email update and sending verification email

matthopson committed:
Merge pull request #7733 from flexion/integration-test-updates

Debugging integration tests

rachaelparris committed:
Merge pull request #7745 from flexion/7574-edit-practitioner-detail-ui

7574: edit practitioner detail UI - pending email and email not provided

mrinsin committed:
Merge pull request #7742 from flexion/7574-verification-modal

7574: Verification modal

mrinsin committed:
updating to new usage of moize

mrinsin committed:
importing eventContext directly from serverless package

mrinsin committed:
updating how moize is imported

klohman committed:
Merge pull request #7743 from flexion/pr62-release-notes

Update release-notes.md

ericsorenson committed:
Merge branch 'develop' into from-usataxcourt

klohman commented:
## PR 62 Release Notes 

**Improvement** Petitioners and practitioners can change the email used for login and service through their DAWSON account. 

**Improvement** Admissions Clerks can change an existing practitioner’s email in DAWSON. This will reflect in all of their cases and be used for login and service. 

**Improvement** Bench Opinions now display in the Today’s Opinions list. 

**Bug Fix** Document title did not display when attaching a draft document to a message. 
* When selecting a draft document to attach to a message, the full document title now displays in the dropdown.

**Bug Fix** Users were being logged out when drafting orders using multiple tabs. 
* When working in multiple tabs, you will not be logged out until there is inactivity in all tabs. 

**Bug Fix** The Processed and Served boxes for Docket and Petitions now show items completed within the last 7 days, sorted newest to oldest. 

**Bug Fix** Petitioner service now displays accurate service preference for IRS attorneys before they are associated with the case.

rachaelparris committed:
don't use serverless-express event context

rachelschneiderman committed:
7574: adding integration tests [skip ci]

ericsorenson committed:
Merge branch 'from-usataxcourt' of https://github.com/flexion/ef-cms into from-usataxcourt

ericsorenson committed:
downgrade moize

rachelschneiderman committed:
7574: Fixing integration tests

rachelschneiderman committed:
7574: Adding a couple more expectations for UI updates

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-update-pracitioner-email-api

ericsorenson committed:
Merge pull request #7744 from flexion/from-usataxcourt

From usataxcourt/staging

ericsorenson committed:
Revert "From usataxcourt/staging"

ericsorenson committed:
Merge pull request #7747 from flexion/revert-7744-from-usataxcourt

Revert "From usataxcourt/staging"

rachelschneiderman committed:
7574: Send email verification link to practitioner user AFTER updating their new email in persistence

rachaelparris committed:
7574: update practitioner user with pending email in correct order

ericsorenson committed:
package upgrades

rachelschneiderman committed:
Merge pull request #7746 from flexion/7574-update-pracitioner-email-api

7574: Update API to handle pending email update and sending verification email

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-dod-integration-tests

rachelschneiderman committed:
7574: Update npm packages

rachelschneiderman committed:
7574: Add test around validation bug

kkoskelin committed:
Devex: don't use array index for key on jsx loops

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-array-index

rachelschneiderman committed:
Merge pull request #7749 from flexion/7574-dod-integration-tests

7574 DOD: Integration test, refactoring

rachelschneiderman committed:
7574: Add tests around updatedEmail and confirmEmail validation

rachelschneiderman committed:
7574: Don't persist confirmEmail and updatedEmail on Practitioner entities to solve form population bug

mrinsin committed:
7574 adding unit tests to new method

mrinsin committed:
Merge pull request #7753 from flexion/7574-clear-update-email-form

7574 clear update email form after submitting email updates

rachaelparris committed:
7574: refactoring test data

matthopson committed:
7717 - PublicCase Entity - add isPaper for correct computation of service pref

rachaelparris committed:
7574: refactoring test setup

kkoskelin committed:
Merge pull request #7750 from flexion/devex-array-index

Devex: better loop keys (no array indexes)

ericsorenson committed:
Merge branch 'develop' into experimental1

ericsorenson committed:
Update package-lock.json

rachaelparris committed:
Merge pull request #7752 from flexion/7574-dod

7574 DOD: npm update, make validation more strict

rachelschneiderman committed:
7474: Update edit practitioner submit to show progress bar

rachelschneiderman committed:
Merge pull request #7755 from flexion/7574-add-progress-spinner

7474: Update edit practitioner submit to show progress bar

matthopson committed:
7717 - Update caseFilter to include isPaper for sealed

matthopson committed:
7717 - Add integration test for service indicator on paper case

matthopson committed:
Merge remote-tracking branch 'origin/develop' into 7717-bug-service-indicator

ericsorenson committed:
Merge branch 'develop' into experimental1

ericsorenson committed:
Merge branch 'experimental1' into develop

kkoskelin committed:
7717: verify `isPaper` is attached to PublicCase entity

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into 7717-bug-service-indicator

rachaelparris committed:
Devex 552: move links into computeds for message details

rachelschneiderman committed:
7574: Show alert success message after navigating back to practitioner detail when updating contact information

rachaelparris committed:
Devex 552: move links into computeds for pending report

rachaelparris committed:
Merge pull request #7760 from flexion/devex-552

Devex 552: move links into computeds for message details

rachaelparris committed:
address PR feedback

rachelschneiderman committed:
Merge pull request #7761 from flexion/7574-ux-feedback

7574: Show alert success message after navigating back to practitioner detail when updating contact information

mrinsin committed:
204 adding more unit tests

kkoskelin committed:
devex: no array indexes for loop keys

more.

kkoskelin committed:
Reintroduce async to fix tests

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-array-index

kkoskelin committed:
devex: fewer array indexes as loop keys

mrinsin committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 204-devex-more-action-test-files

rachaelparris committed:
Devex 552: move links from jsx to computeds for message document

kkoskelin committed:
fixing typo

matthopson committed:
7717 - Close socket after integration test

matthopson committed:
Merge remote-tracking branch 'origin/develop' into 7717-bug-service-indicator

rachelschneiderman committed:
7574: Adding tests around bug [skip ci]

matthopson committed:
Merge pull request #7757 from flexion/7717-bug-service-indicator

7717 bug service indicator

kkoskelin committed:
7574: detect new accounts

rachaelparris committed:
Merge pull request #7764 from flexion/devex-552

Devex 552: move links from jsx into computeds for pending report and message document

rachelschneiderman committed:
7574: Updating and adding tests, rethrow error after logging

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-ux-feedback

rachaelparris committed:
Devex 552: use computed instead of jsx for edit statistic link

rachaelparris committed:
Devex 552: move more links from jsx into computeds

rachaelparris committed:
Devex 744: move case type descriptions into constants

rachaelparris committed:
Devex 719: remove instances of  in action/interactors and instead have the router make that call

codyseibert committed:
7748 - only try to generate the coversheet on uploaded files

codyseibert committed:
7748 - only try to generate the coversheet on uploaded files

rachaelparris committed:
Merge pull request #7767 from flexion/devex-744

Devex 744: move case type descriptions into constants

codyseibert committed:
7748 - fixing tests

codyseibert committed:
7748 - fixing tests

codyseibert committed:
7748 - fixing tests

codyseibert committed:
7748 - cleaning up consolidated code

rachelschneiderman committed:
7574: Clear spinner after update practitioner info is successful

rachelschneiderman committed:
7574: Check if email has been updated on the practitioner form to determine if a new cognito account is necessary

rachelschneiderman committed:
7574: Emails working as expected, need to refactor

rachelschneiderman committed:
7574: Refactoring, close web socket after test ends

rachelschneiderman committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 7574-ux-feedback

rachaelparris committed:
Merge pull request #7766 from flexion/devex-552

Devex 552: move more links from jsx into computeds

rachaelparris committed:
Merge pull request #7768 from flexion/devex-719

Devex 719: remove instances of decodeURIComponent in action/interactors and instead have the router make that call

kkoskelin committed:
Merge branch 'develop' of github.com:flexion/ef-cms into devex-array-index

# Conflicts:
#	web-client/src/views/CaseDetail/Statistics.jsx

mrinsin committed:
Merge branch 'develop' of https://github.com/flexion/ef-cms into 204-devex-more-action-test-files

mrinsin committed:
fixing typo

mrinsin committed:
Merge pull request #7763 from flexion/204-devex-more-action-test-files

204 adding more unit tests

rachaelparris committed:
Merge pull request #7769 from flexion/7748-only-add-coversheet-to-uploaded-docs

7748 only add coversheet to uploaded docs

rachaelparris committed:
Devex 618: check for AWS credentials in environment destroy script

rachelschneiderman committed:
7574: Fix integration test, don't cache pratitioner detail

kkoskelin committed:
Merge pull request #7762 from flexion/devex-array-index

Devex array index

rachaelparris committed:
Merge pull request #7774 from flexion/devex-618

Devex 618: check for AWS credentials in environment destroy script

rachelschneiderman committed:
Merge pull request #7775 from flexion/7574-ux-feedback

7574: UX feedback

rachaelparris committed:
Devex 817/836: consolidate query string operations to all use 'qs' module

rachaelparris committed:
Devex 817: strip first character off for parsing in query string decoder

rachaelparris committed:
Merge pull request #7776 from flexion/devex-817

Devex 817/836: consolidate query string operations to all use 'qs' module

rachaelparris committed:
Devex 866: remove case.userId

mrinsin committed:
204 action tests

rachaelparris committed:
remove comment

rachaelparris committed:
Devex 886: remove validationName in favor of entityName

rachaelparris committed:
Merge pull request #7777 from flexion/devex-866

Devex 866: remove case.userId

rachaelparris committed:
Merge branch 'develop' into devex-886

rachaelparris committed:
test fix

rachaelparris committed:
Merge pull request #7778 from flexion/devex-886

Devex 886: remove validationName in favor of entityName

rachaelparris committed:
Devex 884: expand entitiesByName and use for validation migration

rachaelparris committed:
Devex 884: remove entityName for getting consolidated cases so necessary data is not stripped out

kkoskelin committed:
devex: avoid array index as key

mrinsin committed:
Merge pull request #7779 from flexion/204-devex-more-action-test-files

Devex 204: adding tests for remaining uncovered actions

rachaelparris committed:
Merge pull request #7780 from flexion/devex-884

Devex 884: expand entitiesByName and use for validation migration

kkoskelin committed:
Merge pull request #7781 from flexion/devex-array-index

devex: avoid array index as key

rachelschneiderman committed:
WIP diff old/updated practitioner to see if only email was updated [skip ci]

matthopson committed:
7574 - check if only email was updated in interactor [skip ci]

rachaelparris committed:
7574: finish writing tests; refactoring

rachaelparris committed:
7574: remove comment

rachaelparris committed:
7574: use a constant

ericsorenson committed:
upgrades

rachaelparris committed:
Merge pull request #7782 from flexion/7574-hide-progress-bar

7574: hide progress bar when only email is updated

rachaelparris committed:
Devex 776: create external document decorator for shared properties

rachaelparris committed:
Devex 776: create court issued document decorator for shared properties

ericsorenson committed:
Merge pull request #7783 from flexion/from-usataxcourt

Upgrades

rachaelparris committed:
Merge pull request #7785 from flexion/devex-776

Devex 776: DRY up external and court issued document nonstandard types

mrinsin committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into pr-62

# Conflicts:
#	web-api/migration-terraform/main/lambdas/migration-segments.js

ericsorenson committed:
fix test imports

mmarcotte commented on shared/src/business/entities/externalDocument/ExternalDocumentBase.js:
```diff
@@ -0,0 +1,18 @@
+  documentTitle: JoiValidationConstants.DOCUMENT_TITLE.optional(),
+  documentType: JoiValidationConstants.STRING.required(),
+};
+
```
While I appreciate the ♻️ reusing here, this makes me think that we can go further and create a base class that the others extend. 🤔 Nothing for now, but perhaps something for the devex backlog to consider.

kkoskelin replied:
Using classes and inheritance isn't a bad idea, although moving away from the decorator pattern will take some time, and I'm uncertain if the time investment will yield tremendous benefits.

mmarcotte commented on shared/src/business/useCases/externalDocument/uploadExternalDocumentsInteractor.js:
```diff
@@ -86,24 +94,33 @@ exports.uploadExternalDocumentsInteractor = async ({
+          applicationContext,
+          documentMetadata,
+        }),
+      docketEntryIdsAdded,
```
Looks like we're not testing anything for this new return value (or caseDetail for that matter) in our test coverage for this file. Can we address that?

kkoskelin replied:
Test has been updated to review the structure of the return value.

mmarcotte replied:
Thank you!

mmarcotte commented on shared/src/business/useCases/generateSignedDocumentInteractor.test.js:
```diff
@@ -28,7 +28,7 @@ describe('generateSignedDocument', () => {
 
   beforeEach(() => {
-    jest.setTimeout(10000);
+    jest.setTimeout(30000);
```
I worry about this being the right solution to fixing these tests. Nothing to change here, but 30 seconds -- where do we go from here?

kkoskelin replied:
I believe we were having some issues with unexplained intermittent failures, and giving them more time (and bumping the container size) seemed to fix them.

mmarcotte commented on shared/src/business/useCases/practitioners/updatePractitionerUserInteractor.js:
```diff
@@ -66,16 +96,51 @@ exports.updatePractitionerUserInteractor = async ({
-    websocketMessagePrefix: 'admin',
-  });
+  }).toRawObject();
+  const oldPractitionerRaw = new Practitioner(oldUserInfo, {
+    applicationContext,
+  }).toRawObject();
+
+  const practitionerDetailDiff = applicationContext
+    .getUtilities()
+    .getAddressPhoneDiff({
+      newData: {
+        ...omit(updatedPractitionerRaw, 'contact'),
+        ...updatedPractitionerRaw.contact,
+      },
+      oldData: {
+        ...omit(oldPractitionerRaw, 'contact'),
+        ...oldPractitionerRaw.contact,
+      },
+    });
+
+  const hasUpdatedEmailOnly = isEqual(
+    Object.keys(practitionerDetailDiff).sort(),
+    ['pendingEmail', 'pendingEmailVerificationToken'],
+  );
+
+  if (!hasUpdatedEmailOnly) {
```
I dig this, but perhaps it could be put into a reusable function and not clutter up this one to only figure out a boolean.

kkoskelin replied:
I've taken the general comparison code into its own function.

mmarcotte commented on shared/src/business/useCases/practitioners/updatePractitionerUserInteractor.test.js:
```diff
@@ -1,46 +1,65 @@
       userId: 'admissionsclerk',
     };
 
+    mockPractitioner = {
```
Why don't we make use of `shared/src/test/mockUsers.js` here?

kkoskelin replied:
This mock practitioner has now been moved to `mockUsers`.

mmarcotte commented on shared/src/business/useCases/practitioners/updatePractitionerUserInteractor.test.js:
```diff
@@ -140,4 +170,136 @@ describe('updatePractitionerUserInteractor', () => {
+      });
+    });
+
+    it("should NOT send the verification email when the user's email is being added for the first time", async () => {
+      mockPractitioner.email = undefined;
+      await updatePractitionerUserInteractor({
+        applicationContext,
+        user: {
+          ...mockPractitioner,
+          confirmEmail: 'free-email-to-use@example.com',
+          updatedEmail: 'free-email-to-use@example.com',
+        },
+      });
+
+      expect(
+        applicationContext.getUseCaseHelpers().sendEmailVerificationLink,
+      ).not.toHaveBeenCalled();
+    });
```
We should still be sending out a Cognito "welcome email" though, correct? Could we test for that?

kkoskelin replied:
You are correct -- but in _that_ case, the system currently relies on cognito configuration alone to send those emails. They are not handled in the JS interactor, and are not something that is testable here.

mmarcotte commented on shared/src/business/useCases/users/updateUserPendingEmailInteractor.js:
```diff
@@ -0,0 +1,67 @@
+const { User } = require('../../entities/User');
+
+/**
+ * updateUserPendingEmailInteractor
```
Could this be more descriptive or informative? It's hard to tell who this function is for. It appears this is only for a Petitioner or Practitioner updating an email address -- not an Admissions Clerk, who also currently happens to have the EMAIL_MANAGEMENT capability.

kkoskelin replied:
JSDocs have been updated here -- please let me know if they hit the mark you're looking for.

mmarcotte replied:
Hate to be a stickler, but perhaps we could rename this from `User` to `PublicUser` in order to distinguish it from Court users. And then add validation that the user object is either Practitioner or Petitioner.

mmarcotte commented on shared/src/business/useCases/users/verifyUserPendingEmailInteractor.js:
```diff
@@ -0,0 +1,211 @@
+const { ROLES } = require('../../entities/EntityConstants');
+const { UnauthorizedError } = require('../../../errors/errors');
+const { User } = require('../../entities/User');
+
```
Really appreciate the other jsDoc work you've done. Could we do it for the functions here as well?

kkoskelin replied:
JSDocs updated here, also.

mmarcotte replied:
ty

mmarcotte commented on shared/src/business/useCases/users/verifyUserPendingEmailInteractor.js:
```diff
@@ -0,0 +1,211 @@
+      petitionerObject.email = user.email;
+
+      // we do this again so that it will convert '' to null
+      const caseEntity = new Case(caseRaw, { applicationContext }).validate();
```
We should probably do the check and the validation for all of the cases BEFORE we attempt any mutations. If this fails for one, we end up in a wonky half-done state.

kkoskelin replied:
The interactor functions have been rewritten to acquire and validate all cases (post-mutation) before attempting persistence back to dynamo.

mmarcotte commented on shared/src/business/useCases/users/verifyUserPendingEmailInteractor.js:
```diff
@@ -0,0 +1,211 @@
+      practitionerObject.email = user.email;
+
+      // we do this again so that it will convert '' to null
+      caseEntity = new Case(caseEntity, { applicationContext });
```
Same here, except this last line appears to be missing the `.validate();`

kkoskelin replied:
validation is now occurring in recent update.

mmarcotte commented on shared/src/business/useCases/users/verifyUserPendingEmailInteractor.js:
```diff
@@ -0,0 +1,211 @@
+    }
+  }
+
+  await applicationContext.getNotificationGateway().sendNotificationToUser({
```
Do we want to do this if the above process fails? I don't think we do.

kkoskelin replied:
a new test has been written to prevent notifications from being sent if the updateCaseAndAssociations helper fails.

mmarcotte commented on shared/src/persistence/dynamo/users/updateUserEmail.js:
```diff
@@ -0,0 +1,31 @@
+      })
+      .promise();
+  } catch (err) {
+    applicationContext.logger.error('Error updating user');
```
Can we include the Username and Email in what we log? What happens with the rest of the application if this fails quietly?

kkoskelin replied:
We'll have a larger discussion (and effort) about error handling, error re-throwing, and amount of details to be logged.

ericsorenson replied:
https://trello.com/c/VkiYGA26

kkoskelin replied:
https://trello.com/c/VkiYGA26/901-revisit-error-handling-specifically-catching-logging-and-not-rethrowing-whats-our-strategy-moving-forward

mmarcotte replied:
thanks all

mmarcotte commented on shared/src/persistence/dynamo/users/updateUserEmail.js:
```diff
@@ -0,0 +1,31 @@
+          },
+        ],
+        UserPoolId: process.env.USER_POOL_ID,
+        Username: response.Username,
```
I think you can use `user.email` here and avoid the first call.

rachaelparris replied:
We follow this pattern in several places, and we just duplicated it again here. Would you be ok with us adding this as a refactoring task? There might be a reason this was done this way, but either way there is an opportunity to DRY up these calls with a helper.

mmarcotte replied:
👌 Thanks @rachaelparris

mmarcotte commented on web-api/src/users/updateUserPendingEmailLambda.js:
```diff
@@ -0,0 +1,17 @@
+      .getUseCases()
+      .updateUserPendingEmailInteractor({
+        applicationContext,
+        ...JSON.parse(event.body),
```
Can we only send this what it needs instead of what it gets?

kkoskelin replied:
When the contract between the client proxy and the interactor changes, this will be one layer which need not change.  I think it is better to simply pass along the entire body, no harm done.

mmarcotte commented on web-api/src/users/verifyUserPendingEmailLambda.js:
```diff
@@ -0,0 +1,17 @@
+      .getUseCases()
+      .verifyUserPendingEmailInteractor({
+        applicationContext,
+        ...JSON.parse(event.body),
```
same

kkoskelin replied:
see previous comment.

mmarcotte commented on web-client/integration-tests/helpers.js:
```diff
@@ -110,6 +110,15 @@ export const getEmailsForAddress = address => {
     applicationContext,
   });
 };
+export const getPendingEmailVerificationTokenForUser = userId => {
```
Hmm this function name appears to be doing more than it describes. Perhaps it should be renamed or more specific in what it `@returns`.

kkoskelin replied:
renamed.

mmarcotte commented on web-client/integration-tests/modifyPractitionerEmail.test.js:
```diff
@@ -0,0 +1,46 @@
+  let caseDetail;
+  test.createdDocketNumbers = [];
+
+  loginAs(test, 'privatePractitioner2@example.com');
```
Mayhaps set `privatePractitioner2@example.com` as a constant so we can just re-use it. Then it could be even more dynamic in the future. Love that I'm swimming in integration tests btw. Thanks for these!

kkoskelin replied:
done.

kkoskelin committed:
pr-61: addressing some comments

kkoskelin committed:
pr-62: more comments

[skip ci]

mmarcotte commented on web-client/src/AppInstanceManager.jsx:
```diff
@@ -0,0 +1,31 @@
+import { sequences, state } from 'cerebral';
+import React from 'react';
+
+export const AppInstanceManager = connect(
```
Would really appreciate a little doc about what this does/why. 🙏

kkoskelin replied:
done.

mmarcotte commented on web-client/src/presenter/actions/getDocumentQCInboxForSectionAction.test.js:
```diff
@@ -4,7 +4,7 @@ import { presenter } from '../presenter-mock';
 
 describe('getDocumentQCInboxForSectionAction', () => {
-  const mockWorkItems = [{ docketEnryId: 1 }, { docketEntryId: 2 }];
+  const mockWorkItems = [{ docketEntryId: 1 }, { docketEntryId: 2 }];
```
This led me to search for `docketEnry`, and I found a few other hits. Could we clean those up, too? 🙏

kkoskelin replied:
done.

mmarcotte commented on web-client/src/presenter/actions/getUserAction.test.js:
```diff
@@ -0,0 +1,44 @@
+import { runAction } from 'cerebral/test';
+
+describe('getUserAction', () => {
+  const mockUser = {
```
Seems like another opportunity to use our shared `mockUsers`

kkoskelin replied:
This mock is just a few lines long, and the simplicity of this action really doesn't warrant importing mocks from elsewhere in this case, IMO.

mmarcotte commented on web-client/src/presenter/computeds/caseDetailHelper.js:
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable complexity */
```
⭐

mmarcotte commented on web-client/src/presenter/sequences/closeVerifyEmailModalAndNavigateToPractitionerDetailSequence.js:
```diff
@@ -0,0 +1,13 @@
+import { setAlertSuccessAction } from '../actions/setAlertSuccessAction';
+import { setSaveAlertsForNavigationAction } from '../actions/setSaveAlertsForNavigationAction';
+
+export const closeVerifyEmailModalAndNavigateToPractitionerDetailSequence = [
```
No need to change anything, but might these compound sequences be broken up into two sequences for readability and reusability?

kkoskelin replied:
I'm not sure I follow.  If anything, the sequence has a rather verbose name -- but the number of actions within is brief and easily read.

mmarcotte commented on web-client/src/views/FileDocument/ViewAllDocumentsDesktop.jsx:
```diff
@@ -42,44 +42,42 @@ export const ViewAllDocumentsDesktop = connect(
+                            link
+                            className="text-left"
+                            key={`${title}-document-${doc.eventCode}`}
+                            onClick={() => {
+                              updateFileDocumentWizardFormValueSequence({
+                                key: 'category',
+                                value: doc.category,
+                              });
+                              updateFileDocumentWizardFormValueSequence({
+                                key: 'documentType',
+                                value: doc.documentType,
+                              });
+                              updateFileDocumentWizardFormValueSequence({
+                                key: 'documentTitle',
+                                value: doc.documentTitle,
+                              });
+                              updateFileDocumentWizardFormValueSequence({
+                                key: 'eventCode',
+                                value: doc.eventCode,
+                              });
+                              updateFileDocumentWizardFormValueSequence({
+                                key: 'scenario',
+                                value: doc.scenario,
+                              });
+                              openCompleteSelectDocumentTypeModalSequence();
+                            }}
```
This is a little unruly. I know it's not a new change, but could we add a devex card to refactor

mmarcotte commented on wikiwiki/PublicAccess.md:
```diff
@@ -6,4 +6,9 @@
 ## Today's Opinions
 * Public users have access to all opinions served today via the Today's Opinion page
-* Opinions in sealed cases are available in the Today's Opinions list
+* Opinions in sealed cases are available in the Today's Opinions list and the public can access the PDF for a  opinions in sealed cases
```
just fix the wee typo "for a  opinions..." => "for opinions..."

kkoskelin replied:
Done

mmarcotte commented on wikiwiki/release-notes.md:
```diff
@@ -1,5 +1,64 @@
+**Bug Fix** Petitioner service now displays accurate service preference for IRS attorneys before they are associated with the case.
+
+## PR 61 Release Notes
+
+### Internal
+**Improvement**: Admissions clerks can edit the profile for a practitioner without an email address.
+* Clerks can make changes to practitioner information for practitioners with or without an email address on record.
+
+**Bug Fix**: Docket clerks will not be prompted to print paper service after completing QC and sending a message.
+* When completing QC and sending a message for a document to be served via paper service, clerks are now prompted immediately to print and mail the necessary documents.
+
+**Bug Fix**: Document titles will now display correctly across the application
+* Document titles were missing any “additional info” text when attaching a document to a message. The full document title now displays in the dropdown to attach a document.
+* Updates to the document title during the QC process were not being reflected when sending a message or being updated accurately on the Docket Record. Any updates made to a document title during the QC process reflect immediately on messages and on the Docket Record.
+
+**Bug Fix**: Electronic signatures are now visible in all PDF previews
+*  Some PDF previews were not properly displaying electronic signatures on documents. Documents uploaded to the system with electronic signatures now display the signature in all PDF previews
+* Parties filing documents on mobile devices do not have access to PDF previews prior to submitting their documents.
+
+**Bug Fix**: Errors when clicking on messages has been resolved
+* Some users were encountering a white screen when clicking on their messages, which required to refresh to view the message. Users are now able to access their messages without refresh directly from their inboxes.   
+
+**Bug Fix**: Completed messages will no longer display in any inbox or In Progress queue
+* Completed messages were still displaying in the section Inbox and in the In Progress tab on the case.
+* When a message is completed, it is no longer display in any individual or section Inboxes, or in the In Progress tab under Case Messages. Completed messages will display in the Completed tabs.
+
+**Bug Fix**: Public docket record now displays accurate event codes and filings and proceedings
+* The public view of the Docket Record was displaying an incorrect event code for lodged documents. All lodged documents now show on the Docket Record with the correct event code of MISCL.
+* The public view of the Docket Record was not displaying the complete entry text under Filings and Proceedings. Parenthetical text (i.e. Attachments, Certificate of Service, Objections) now display as part of the Filings and Proceedings for all views of the Docket Record.
+
+### External / Public  
+**Bug Fix**: Electronic signatures are now visible in all PDF previews
+*  Some PDF previews were not properly displaying electronic signatures on documents. Documents uploaded to the system with electronic signatures now display the signature in all PDF previews
+* Parties filing documents on mobile devices do not have access to PDF previews prior to submitting their documents.
+
+**Bug Fix**: Public docket record now displays accurate event codes and filings and proceedings
+* The public view of the Docket Record was displaying an incorrect event code for lodged documents. All lodged documents now show on the Docket Record with the correct event code of MISCL.
+* The public view of the Docket Record was not displaying the complete entry text under Filings and Proceedings. Parenthetical text (i.e. Attachments, Certificate of Service, Objections) now display as part of the Filings and Proceedings for all views of the Docket Record.
+
+
```
❤️

kkoskelin committed:
pr-62: addressing comments for verifyUserPendingEmailInteractor

kkoskelin committed:
pr-62: addressing comments

kkoskelin committed:
pr-62: set mock return value just once

kkoskelin committed:
pr-62: investigating integration failures [skip ci] WIP

kkoskelin committed:
pr-62: fixing implementation back to previous persistence method via map

kkoskelin committed:
pr-62 fixing linting issue

not caught by husky hooks -- not sure why.

kkoskelin committed:
pr-62: integration test fixes

changed args to test components, missed some. add'l logging.

kkoskelin committed:
pr-62: filter out cases not associated with user

filter them instead of throwing exception. continue processing remaining cases and log an error.
